File: FelipezConde/QuestionnaireLanguage/AST/ParseTreeVisitors/TypeVisitor.cs
using Grammar;


namespace AST.ParseTreeVisitors
{
    public class TypeVisitor : QLMainBaseVisitor<Types.Type>
    {
        public override Types.Type VisitBoolType(QLMainParser.BoolTypeContext context)
        {
            return new Types.BoolType();
        }
        public override Types.Type VisitStringType(QLMainParser.StringTypeContext context)
        {
            return new Types.StringType();
        }
        public override Types.Type VisitIntType(QLMainParser.IntTypeContext context)
        {
            return new Types.IntType();
        }
    }
}

File: SanderMark/QL/src/com/form/language/ast/type/TypeVisitor.java
package com.form.language.ast.type;

public interface TypeVisitor<T> {

	    public T visitBoolType(BoolType btype);
	    public T visitIntType(IntType itype);
	    public T visitStringType(StringType stype);
	    public T visitErrorType(ErrorType etype);
}
